chainer/distributions/multivariate_normal.py

```diff
@@ -0,0 +1,198 @@
+import chainer
+from chainer.backends import cuda
+from chainer import distribution
+from chainer.functions.array import broadcast
+from chainer.functions.array import expand_dims
+from chainer.functions.array import moveaxis
+from chainer.functions.array import repeat
+from chainer.functions.array import squeeze
+from chainer.functions.array import stack
+from chainer.functions.array import swapaxes
+from chainer.functions.array import where
+from chainer.functions.math import basic_math
+from chainer.functions.math import exponential
+from chainer.functions.math import matmul
+from chainer.functions.math import sum as sum_mod
+from chainer import utils
+from chainer.utils import type_check
+import math
+import numpy
+
+try:
+    import scipy.linalg
+    available_cpu = True
+except ImportError as e:
+    available_cpu = False
+    _import_error = e
+
+ENTROPYC = 0.5 * math.log(2 * math.pi * math.e)
+LOGPROBC = - 0.5 * math.log(2 * math.pi)
+
+
+class TriangularInv(chainer.function_node.FunctionNode):
+
+    def __init__(self, lower):
+        self._lower = lower
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 1)
+        a_type, = in_types
+        type_check.expect(a_type.dtype == numpy.float32)
+        # Only 2D array shapes allowed
+        type_check.expect(a_type.ndim == 2)
+        # Matrix inversion only allowed for square matrices
+        type_check.expect(a_type.shape[0] == a_type.shape[1])
+
+    def forward_cpu(self, inputs):
+        self.retain_outputs((0,))
+        if not available_cpu:
+            raise ImportError("SciPy is not available. Forward computation"
+                              " of triangular_inv in CPU can not be done." +
+                              str(_import_error))
+        x, = inputs
+        invx = scipy.linalg.solve_triangular(x, numpy.eye(len(x)),
```

Comment:
How about giving `dtype` to `numpy.eye`?  (Then the casting by `force_array` could be avoided.)